src/main/scala/org/broadinstitute/dsde/firecloud/service/DataUseRestrictionSupport.scala

```diff
@@ -58,6 +61,61 @@ trait DataUseRestrictionSupport extends LazyLogging {
 
   }
 
+  /**
+    * Looks at a DuosDataUse object (likely previously retrieved from ORSP) and translates it into a
+    * set of structured Data Use Restriction attributes.
+    *
+    * @param duosDataUse
+    * @return
+    */
+  def generateStructuredUseRestrictionAttribute(duosDataUse: DuosDataUse, ontologyDAO: OntologyDAO): Map[AttributeName, Attribute] = {
+    // these are straightforward mappings
+    val gru = duosDataUse.generalUse.map ( AttributeName.withLibraryNS("GRU") -> AttributeBoolean(_) )
+    val hmb = duosDataUse.hmbResearch.map ( AttributeName.withLibraryNS("HMB") -> AttributeBoolean(_) )
+    val rspd = duosDataUse.vulnerablePopulations.map ( AttributeName.withLibraryNS("RS-PD") -> AttributeBoolean(_) )
+    val ncu = duosDataUse.commercialUse.map ( AttributeName.withLibraryNS("NCU") -> AttributeBoolean(_) )
+    val nmds = duosDataUse.methodsResearch.map ( AttributeName.withLibraryNS("NMDS") -> AttributeBoolean(_) )
+
+    // FC supports both NCU and NPU; DUOS only supports one (NCU)
+    // val npu = duosDataUse.commercialUse.map ( AttributeName.withLibraryNS("NPU") -> AttributeBoolean(_) )
+
+    // DUOS represents NAGR and NCTRL as strings ("yes"|"no")
+    val nagr = duosDataUse.aggregateResearch match {
+      case Some(s) if "yes".equals(s.toLowerCase) => Some(AttributeName.withLibraryNS("NAGR") -> AttributeBoolean(true))
```

Comment:
(Optional) It might be nice to make the string literals in this function into constants, grouped into objects for FC and DUOS.

Motivating example: I feel like we could at some point make a mistake between `"yes"`/`"no"` (DUOS) and `"Yes"`/`"No"` (FC)